ui/raidboss/data/07-dt/dungeon/meso-terminal.ts

```diff
@@ -0,0 +1,405 @@
+    // The best case for finding the locations is to find all the Preserved Terrors,
+    // then filter down which one is in a safe position in front.
+    {
+      id: 'Meso Terminal Immortal Remains First Bombardment Collect',
+      type: 'GainsEffect',
+      netRegex: { effectId: '9F8', capture: false },
+      // The Preserved Terror adds are used for more mechanics than just Bombardment.
+      // Unfortunately, we don't have any good way to separate out
+      // which ones are used for which mechanics.
+      // The tethers for Bombardment are also used on Electray
+      // and Keraunography, so we have to be very careful how we find the adds.
+      condition: (data) => {
+        // The first Memory of the Storm cast immediately precedes Bombardment.
+        return data.seenFirstStorm && !data.seenFirstBombardment;
+      },
+      suppressSeconds: 1,
+      promise: async (data) => {
+        const callData = await callOverlayHandler({
+          call: 'getCombatants',
+        });
+        if (!callData.combatants.length)
+          return;
+        const terrors = callData.combatants.filter((actor) => actor.BNpcID === 18624);
+        data.safeTerror = findNorthTerror(terrors);
+      },
+    },
+    {
+      id: 'Meso Terminal Immortal Remains Bombardment No Knockback',
+      type: 'GainsEffect',
+      netRegex: { effectId: '9F8', capture: false },
+      condition: (data) => data.seenFirstStorm && !data.seenFirstBombardment,
+      delaySeconds: 1,
+      suppressSeconds: 1,
+      alertText: (data, _matches, output) => {
+        if (data.safeTerror === undefined)
+          return output.unknown!();
+        const safeDir = data.safeTerror.PosX < 0 ? 'west' : 'east';
+        return output.staticBombardment!({ safe: output[safeDir]!() });
+      },
+      run: (data) => {
+        data.seenFirstBombardment = true;
+        data.safeTerror = undefined;
+      },
+      outputStrings: {
+        staticBombardment: {
+          en: 'Go ${safe}; Avoid small AoE',
+        },
+        west: Outputs.west,
+        east: Outputs.east,
+        unknown: Outputs.unknown,
+      },
+    },
+    {
+      id: 'Meso Terminal Immortal Remains Bombardment Knockback',
+      type: 'Ability',
+      netRegex: { id: 'AB29', source: 'Immortal Remains', capture: false }, // Impression
+      condition: (data) => data.seenFirstBombardment,
+      delaySeconds: 1,
+      suppressSeconds: 1,
+      promise: async (data) => {
+        const callData = await callOverlayHandler({
+          call: 'getCombatants',
+        });
+        if (!callData.combatants.length)
+          return;
+        const terrors = callData.combatants.filter((actor) => actor.BNpcID === 18624);
+        data.safeTerror = findNorthTerror(terrors);
+      },
+      alertText: (data, _matches, output) => {
+        if (data.safeTerror === undefined)
+          return output.unknown!();
+        const safeDir = data.safeTerror.PosX < 0 ? 'northwest' : 'northeast';
```

Comment:
I think this is outputting the side where the terrors are and the output needs to be flipped.

Reply:
Also interestingly, I tried to replay my log through raidemulator and it got '???' for a call here whereas in game it did call but incorrect side.

Reply:
Hmm, that looks like I overlooked an edge case in `findSafeTerror()`. The Y position should not be run through `abs()`, since we explicitly want only negative (north) values for Y.

Re-running through the raid emulator with that fix, it correctly calls left first, right second, and then ??? on the third, which I suspect is an artifact of the emulator rather than an actual issue. I'll try this next time I run.

Reply:
Thanks for confirming that and sorry to put you to the trouble of tracking down fixes for it.